bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/RawRockerToOnOffProfile.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2014,2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.smarthome.core.thing.internal.profiles;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.CommonTriggerEvents;
+import org.eclipse.smarthome.core.thing.profiles.ProfileCallback;
+import org.eclipse.smarthome.core.thing.profiles.ProfileTypeUID;
+import org.eclipse.smarthome.core.thing.profiles.SystemProfiles;
+import org.eclipse.smarthome.core.thing.profiles.TriggerProfile;
+import org.eclipse.smarthome.core.types.State;
+
+/**
+ * The {@link RawRockerToOnOffProfile} transforms rocker switch channel events into ON and OFF commands.
+ *
+ * @author Jan Kemmler - Initial contribution
+ */
+public class RawRockerToOnOffProfile implements TriggerProfile {
```

Comment:
I assume this profile is a good candidate for a configuration (up = on or up = off).

Reply:
This is a complicated topic. I have been thinking about profiles and their configuration.
This whole thing quickly leads to philosophical issues about where to solve the complexity issue with many different combinations of items and channels and possible translations between events and commands.
As Kai said in another post, there is also the question of having a toggle profile work on a rawrocker channel, which is another example where you now start doing tons of configuration for profiles, so that you can use them for more channel/item combinations.

The more configuration options and fewer individual profiles we use, the more complex the definition of items in *.items files will become.

The configuration for a RawRockerToDimmerProfile for example would then contain parameters for initial delay time, interval time for increasing and decreasing and definition of which event triggers up and which event triggers down. This also means that the user that wants to use this profile also has to know exactly which events each channel emits.

On the other hand, if we split these functionalities into more different profiles, the user has a longer list of different profiles from which to choose the correct one. This is also some added complexity, but it seems a bit less to me.
What do you think?
Do we want to prefer more configuration options over more profiles?

In this specific example with mapping the events upside-down, I don't even know that we really even want to have this functionality. It seems relatively specific and can easily be done via rules, if someone wants to solve the problem of having up and down exchanged.

Also: I have no idea how configurations for profiles work, is there some documentation or an example on how to do it?

Reply:
There is no need that you add configuration support in this PR.
It has been just a comment WRT the forum entry about profile configuration.

For such simple stuff as swap up / down, I would like to see a configuration some time.
If I want to combine a rocker with a light, I don't want to write a rule, just because I want to switch the on / off use case. Sure, rules are an option, but isn't the intention of profiles, that we don't need to create rules for such normal use cases.

Reply:
Okay, I will add this at some point. I am still curious, however, how profile configurations work. Got a link?

Reply:
They are configured at the links, not at the items (there is no support for it yet in the the Paper UI though). And indeed, I got a link (literally 😄 ): 
https://github.com/eclipse/smarthome/blob/36c9e2717e4f89cf60a0d885e7ecc6069b338f48/bundles/model/org.eclipse.smarthome.model.thing.tests/src/org/eclipse/smarthome/model/thing/tests/GenericItemChannelLinkProviderJavaTest.java#L164-L226

Within the profiles you then can get the configuration object via the `ProfileContext`.